### PR TITLE
Add support for tz2, tz3, kt1 addresses

### PR DESCRIPTION
**Pull request by keefertaylor**
## Description

Add support for tz2, tz3, and KT1 addresses.

## Testing instructions

Run bootstrap.sh. Added new tests for additional address formats.

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [ X ] Prefix PR title with `[WIP]` if necessary.
- [ X ] Add tests to cover changes as needed.
- [ X ] Update documentation as needed.

### CODE REVIEW

File: tests/Tezos/AddressTests.cpp [hunk 1]
```diff
@@ -60,8 +60,17 @@ TEST(TezosAddress, isInvalid) {
...
-    ASSERT_TRUE(Address::isValid(address));
+    std::array<std::string, 4> validAddresses {        
+        "tz1Yju7jmmsaUiG9qQLoYv35v5pHgnWoLWbt",
+        "tz2PdGc7U5tiyqPgTSgqCDct94qd6ovQwP6u",
+        "tz3VEZ4k6a4Wx42iyev6i2aVAptTRLEAivNN",
+        "KT1VsSxSXUkgw6zkBGgUuDXXuJs9ToPqkrCg"  
+    };
+
+    for (std::string address : validAddresses) {
+      std::cout << address << std::endl;
```
Comment (hewigovens) [inline tests/Tezos/AddressTests.cpp]: please remove debug output

File: tests/Tezos/AddressTests.cpp [hunk 2]
```diff
@@ -60,8 +60,17 @@ TEST(TezosAddress, isInvalid) {
...
-    auto address = "tz1d1qQL3mYVuiH4JPFvuikEpFwaDm85oabM";
-    ASSERT_TRUE(Address::isValid(address));
+    std::array<std::string, 4> validAddresses {        
+        "tz1Yju7jmmsaUiG9qQLoYv35v5pHgnWoLWbt",
+        "tz2PdGc7U5tiyqPgTSgqCDct94qd6ovQwP6u",
+        "tz3VEZ4k6a4Wx42iyev6i2aVAptTRLEAivNN",
+        "KT1VsSxSXUkgw6zkBGgUuDXXuJs9ToPqkrCg"  
+    };
+
+    for (std::string address : validAddresses) {
```
Comment (hewigovens) [inline tests/Tezos/AddressTests.cpp]:
```cpp
for (auto& address : validAddresses)
```

File: src/Tezos/Address.cpp [hunk 3]
```diff
@@ -24,12 +24,18 @@ bool Address::isValid(const std::string& string) {
     }
 
     // verify prefix
-    std::array<byte, 3> prefix{6, 161, 159};
-    if (!std::equal(prefix.begin(), prefix.end(), decoded.begin())) {
-        return false;
+    std::array<byte, 3> tz1Prefix{6, 161, 159};
```
Comment (hewigovens) [inline src/Tezos/Address.cpp]: can we make them constants? I saw `prefix{6, 161, 159}` in `Tezos::Address` too

File: tests/Tezos/AddressTests.cpp (hunk 2 above)
Comment (keefertaylor) [reply tests/Tezos/AddressTests.cpp]: Fixed above test to use consistent style as well.

File: src/Tezos/Address.cpp (hunk 3 above)
Comment (keefertaylor) [reply src/Tezos/Address.cpp]: Sure. 

I refactored forging.cpp as well to not use these constants so we don't need to extern. 

There are forge_tzX test cases in AddressTests.cpp which validate this code works.

**Pull request merged**